packages/eslint-plugin/src/rules/consistent-indexed-object-style.ts

```diff
@@ -113,14 +119,37 @@ export default createRule<Options, MessageIds>({
             return;
           }
 
+          const indexParam = params[0];
+
+          const shouldAutoFix =
+            indexParam.type === AST_NODE_TYPES.TSStringKeyword ||
+            indexParam.type === AST_NODE_TYPES.TSNumberKeyword ||
+            indexParam.type === AST_NODE_TYPES.TSSymbolKeyword;
+
+          const fix: TSESLint.ReportFixFunction = fixer => {
+            const key = context.sourceCode.getText(params[0]);
+            const type = context.sourceCode.getText(params[1]);
+            return fixer.replaceText(node, `{ [key: ${key}]: ${type} }`);
+          };
+
           context.report({
             node,
             messageId: 'preferIndexSignature',
-            fix(fixer) {
-              const key = context.sourceCode.getText(params[0]);
-              const type = context.sourceCode.getText(params[1]);
-              return fixer.replaceText(node, `{ [key: ${key}]: ${type} }`);
-            },
+            fix: shouldAutoFix
```

Comment:
(not a blocker) wondering whether getOrSuggest can be used here for some visual simplification?

https://github.com/typescript-eslint/typescript-eslint/blob/5b2ebcdea9011a6de7e657118874ae85c08d5a0e/packages/eslint-plugin/src/util/getFixOrSuggest.ts#L1-L21

Reply:
Nice, it can, though it would require an `eslint-disable` line similar to this one, as it's the only suggestion the rule has: https://github.com/typescript-eslint/typescript-eslint/blob/d24a82854d06089cbd2a8801f2982fd4781f3701/packages/eslint-plugin/src/rules/return-await.ts#L47-L48

I've updated the PR to use `getOrSuggest`, and I think that even though it requires the disable comment, it helps simplify the logic.